tests/saml/test_provider.py

```diff
@@ -350,10 +360,126 @@ def test_authentication_document(self, _, identity_providers, expected_result):
                     complete=True,
                 ),
             ),
+            (
+                "subject_has_unique_name_id_but_use_of_name_id_is_switched_off_using_integer_literal",
```

Comment:
Again, representing boolean configuration settings as binary zeros and ones makes the code more complex since we don't whether we're going to end up with an integer zero or one or their string version.